### ISSUE TITLE
Use buttons instead of links in ConfiguratorGroupMenuComponent

### ISSUE THREAD

**Issue by Platonn** [SAP/spartacus#12089]
Correlate to #10525

We're should not use anchor elements for things that are sematically buttons (they don't have `href` attribute, but invoke JS logic). But it's the case for `ConfiguratorGroupMenuComponent`. For example:
```typescript
        <a
          class="cx-menu-back-link"
          role="link"
          tabindex="0"
          (click)="navigateUp()"
          (keyup)="navigateUpOnEnter($event)"
        >
```
That forces us to implement custom event handler (keyup), which is a redundant logic.
Or to implement redundant custom event handler to avoid scrolling down for spacebar `(keydown)="preventScrollingOnSpace($event)"`

Instead we should use a `<button>` which is semantically correct. Btw. it also supports space bar natively. So we can also get rid off all unnecessary custom event handlers.

Changing DOM elements from `<a>` to `<button>` is a breaking change in the DOM. So should be done in a major release.

Make possible to navigation via Arrow keys within the group menu: [w3.org](https://www.w3.org/TR/2019/NOTE-wai-aria-practices-1.1-20190814/examples/tabs/tabs-2/tabs.html)

**Comment by Larisa-Staroverova** [comment]
Group navigation via Arrao keys: [w3.org](https://www.w3.org/TR/2019/NOTE-wai-aria-practices-1.1-20190814/examples/tabs/tabs-2/tabs.html)
